tarodb/sqlerrors.go

```diff
@@ -0,0 +1,64 @@
+package tarodb
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/jackc/pgconn"
+	"github.com/jackc/pgerrcode"
+	"modernc.org/sqlite"
+	sqlite3 "modernc.org/sqlite/lib"
+)
+
+// ParseSqlError attempts to interpret a given error as a database agnostic SQL
+// error.
+func ParseSqlError(err error) error {
+	// Attempt to interpret the error as a sqlite error.
+	var sqliteErr *sqlite.Error
+	if errors.As(err, &sqliteErr) {
+		return parseSqliteError(sqliteErr)
+	}
+
+	// Attempt to interpret the error as a postgres error.
+	var pqErr *pgconn.PgError
+	if errors.As(err, &pqErr) {
+		return parsePostgresError(pqErr)
+	}
+
+	// Indicate that the error could not be parsed.
+	return nil
+}
+
+// parsePostgresError attempts to parse a sqlite error as a database agnostic
+// SQL error.
+func parseSqliteError(sqliteErr *sqlite.Error) error {
+	// Handle unique constraint violation error.
+	if sqliteErr.Code() == sqlite3.SQLITE_CONSTRAINT_UNIQUE {
```

Comment:
Since we might add more error codes in the future, this could be a `switch` statement over `sqliteErr.Code()` with the default returning the error.